checker/checker.go

```diff
@@ -262,24 +262,55 @@ func (c *Checker) Process(ctx context.Context, pr chan pb.ProcessResult) {
 		errs = append(errs, unit.NewProcessError(err))
 	} else if !result.Summary.Passed {
 		errs = append(errs, unit.NewProcessError(errors.New("check was failed, please see detail")))
+		warnLeft, errLeft := c.warnCnt, c.errCnt
 
 		// remove success result if not pass
 		results := result.Results[:0]
-		var warnCnt int64 = 0
-		var errCnt int64 = 0
 		for _, r := range result.Results {
-			switch r.State {
-			case check.StateFailure:
-				if errCnt < c.errCnt {
-					errCnt++
+			if r.State == check.StateSuccess {
+				continue
+			}
+
+			// handle results without r.Errors
+			if len(r.Errors) == 0 {
+				switch r.State {
+				case check.StateWarning:
+					if warnLeft == 0 {
```

Comment:
To handle `c.warnCnt == 0`
```suggestion
					if warnLeft <= 0 {
```

Reply:
we can handle `warnLeft == 0` in this implementation 🤔

Reply:
But we didn't check this now...